gurock/trcli
# Update case data using automation_id

## Issues being resolved: 
- https://github.com/gurock/trcli/issues/70
- https://github.com/gurock/trcli/issues/91
- https://github.com/gurock/trcli/issues/102
- https://github.com/gurock/trcli/issues/103

### Solution description
Legacy logic was matching test cases by name, which should not be happening, so we are removing such logic and matching by automation_id only.

### Changes
- Test cases are now being matched by `automation_id` when updating `api_data_provider.py` data
- `automation_id` is now stripped of leading and trailing whitespaces on `dataclass_testrail.py`

### Potential impacts
N/A

### Steps to test
- Execute the CLI to import test cases and results
- Change the name of test cases in TestRail
- Execute the CLI again with same data
- Test cases should be successfully matched

### PR Tasks
